src/coreComponents/physicsSolvers/solidMechanics/SolidMechanicsStateReset.hpp

```diff
@@ -0,0 +1,110 @@
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SolidMechanicsStateReset.hpp
+ */
+
+#ifndef SRC_CORECOMPONENTS_PHYSICSSOLVERS_SOLIDMECHANICS_SOLIDMECHANICSSTATERESET_HPP_
+#define SRC_CORECOMPONENTS_PHYSICSSOLVERS_SOLIDMECHANICS_SOLIDMECHANICSSTATERESET_HPP_
+
+#include "events/tasks/TaskBase.hpp"
+
+namespace geosx
+{
+
+class SolidMechanicsLagrangianFEM;
+
+/**
+ * @class SolidMechanicsStateReset
+ *
+ * Task class allowing for various "reset" actions on state variables, useful for pre-
+ * and post-equilibration actions. Currently, this class gives the user the option
+ * to reset displacements to zero and enable/disable inelastic behavior. A common
+ * workflow might be:
+ * (1) resetDisplacments = false, disableInelasticity = true,
+ * (2) solve an equilibration problem for initial conditions,
+ * (3) resetDisplacements = true, disablePlasticity = false,
+ * (4) run a normal simulation
+ *
+ */
+class SolidMechanicsStateReset : public TaskBase
+{
+public:
+
+  /**
+   * @brief Constructor for the state reset class
+   * @param[in] name the name of the task coming from the xml
+   * @param[in] parent the parent group of the task
+   */
+  SolidMechanicsStateReset( const string & name,
+                            Group * const parent );
+
+  /// Destructor for the class
+  ~SolidMechanicsStateReset() override;
+
+  /// Accessor for the catalog name
+  static string catalogName() { return "SolidMechanicsStateReset"; }
+
+  /**
+   * @defgroup Tasks Interface Functions
+   *
+   * This function implements the interface defined by the abstract TaskBase class
+   */
+  /**@{*/
+
+  virtual bool execute( real64 const time_n,
+                        real64 const dt,
+                        integer const cycleNumber,
+                        integer const eventCounter,
+                        real64 const eventProgress,
+                        DomainPartition & domain ) override;
+
+  /**@}*/
+
+private:
+
+  /**
+   * @struct viewKeyStruct holds char strings and viewKeys for fast lookup
+   */
+  struct viewKeyStruct
+  {
+    /// String for the solid solver name
+    constexpr static char const * solidSolverNameString() { return "solidSolverName"; }
+    /// String for the displacement reset flag
+    constexpr static char const * resetDisplacementsString() { return "resetDisplacements"; }
+    /// String for the inelasticity flag
+    constexpr static char const * disableInelasticityString() { return "disableInelasticity"; }
+  };
+
+
+  void postProcessInput() override;
+
+  /// Name of the solid mechanics solver
+  string m_solidSolverName;
+
+  /// Pointer to the solid mechanics solver
+  SolidMechanicsLagrangianFEM * m_solidSolver;
```

Comment:
Yes, I confirm that it is important to have a `SolidMechanicsLagrangianFEM` here, because `m_solidSolver` has to point to the object that registers the displacements and that knows the target regions where they have been registered.